### PR TITLE
[OTel] Refactor logging instrumentation

### PR DESCRIPTION
**Pull request by GSVarsha**
### Changes:
- refactor logging instrumentation
- adapt logging unit tests to OTel spec

### Unit test results:

Screenshot 2024-07-23 at 2 49 53 PM

### CODE REVIEW

File: src/instana/instrumentation/logging.py [hunk 1]
```diff
@@ -7,11 +7,11 @@
...
 
-from ..log import logger
-from ..util.traceutils import get_tracer_tuple, tracing_is_off
+from instana.log import logger
+from instana.util.traceutils import get_tracer_tuple, tracing_is_off
 
 
-@wrapt.patch_function_wrapper('logging', 'Logger._log')
+@wrapt.patch_function_wrapper("logging", "Logger._log")
 def log_with_instana(wrapped, instance, argv, kwargs):
```
Comment (pvital) [inline src/instana/instrumentation/logging.py:15]: I am missing type hints in the method signature.

File: tests/clients/test_logging.py [hunk 2]
```diff
@@ -3,74 +3,77 @@
...
 
-    def test_no_span(self):
-        with tracer.start_active_span('test'):
-            self.logger.info('info message')
+    def test_no_span(self) -> None:
+        with tracer.start_as_current_span("test"):
+            self.logger.info("info message")
 
         spans = self.recorder.queued_spans()
         self.assertEqual(1, len(spans))
```
Comment (pvital) [inline tests/clients/test_logging.py:28]: Let's use the [standard Python assert](https://docs.pytest.org/en/7.1.x/how-to/assert.html) expression from now on. 
Could you please update this test file to use it? Use always the pattern `assert value_to_evaluate <operation> expected_value`. 
```suggestion
        assert len(spans) == 1
```

**Review by pvital — CHANGES_REQUESTED** [review]
What's the coverage percentile of `instrumentation/logging.py` after your changes? Can you create new unit tests to make it at least 95%?

File: src/instana/instrumentation/logging.py (hunk 1 above)
Comment (GSVarsha) [reply src/instana/instrumentation/logging.py]: Done

File: tests/clients/test_logging.py (hunk 2 above)
Comment (GSVarsha) [reply tests/clients/test_logging.py]: Done

**Comment by GSVarsha** [comment]
> What's the coverage percentile of `instrumentation/logging.py` after your changes? Can you create new unit tests to make it at least 95%?

Its 100% now!

```
Name                                                      Stmts   Miss  Cover   Missing

-----------------------------------------------------------------------------------------------------
src/instana/instrumentation/logging.py                      34      0   100%
```

**Review by pvital — APPROVED** [review]
It looks good to me.

**Pull request merged**